cobbler/tftpgen.py

```diff
@@ -531,19 +533,22 @@ def get_profiles_menu(self, menu, metadata, arch: enums.Archs):
                         current_menu_items[boot_loader] = ""
                     current_menu_items[boot_loader] += contents
 
+                    if boot_loader not in menu_labels:
+                        menu_labels[boot_loader] = []
+
                     # iPXE Level menu
                     if boot_loader == "ipxe":
-                        current_menu_items[boot_loader] += "\n"
-                        if "ipxe" not in menu_labels:
-                            menu_labels["ipxe"] = []
+                        display_name = profile.name
+                        if profile.display_name and profile.display_name != "":
+                            display_name = profile.display_name
                         menu_labels["ipxe"].append(
-                            {"name": profile.name, "display_name": profile.name}
+                            {"name": profile.name, "display_name": display_name}
                         )
 
         metadata["menu_items"] = current_menu_items
         metadata["menu_labels"] = menu_labels
 
-    def get_images_menu(self, menu, metadata, arch: enums.Archs):
+    def get_images_menu(self, menu: Menu, metadata: dict, arch: enums.Archs):
```

Comment:
See comment above for cyclic imports

```suggestion
    def get_images_menu(self, menu, metadata: dict, arch: enums.Archs):
```